src/main/java/net/datafaker/service/RandomService.java

```diff
@@ -141,6 +145,84 @@ public Random getRandomInternal() {
         return random;
     }
 
+    /**
+     * Returns a weighted random element from the given list, where each element is represented as a Map
+     * containing a weight and the corresponding value.
+     *
+     * @param items A list of maps, where each map contains:
+     *              - weight: A Double representing the weight of the element, influencing its selection probability.
+     *              - value: The actual element of type T to be randomly selected based on its weight.
+     * @param <T> The type of the element to be selected from the list. The value associated with the weight can be of any type.
+     * @return A randomly selected element based on its weight.
+     * @throws IllegalArgumentException if the list is null, empty, if any item in the list is null or empty,
+     *                                  if any weight is null, non-positive, NaN, negative zero, infinite,
+     *                                  or if the item does not contain 'weight' or 'value' keys.
+     */
+    public <T> T weightedArrayElement(List<Map<String, Object>> items) {
+        validateItemsList(items);
+
+        double totalWeight = calculateTotalWeight(items);
+        double randomValue = random.nextDouble() * totalWeight;
+
+        return selectWeightedElement(items, randomValue);
+    }
+
+    private void validateItemsList(List<Map<String, Object>> items) {
+        if (items == null || items.isEmpty()) {
+            throw new IllegalArgumentException("weightedArrayElement expects a non-empty list");
+        }
+        items.forEach(this::validateItem);
+    }
+
+    private void validateItem(Map<String, Object> item) {
+        if (item == null || item.isEmpty()) {
+            throw new IllegalArgumentException("Item cannot be null or empty");
+        }
+        if (!item.containsKey(WEIGHT_KEY) || !item.containsKey(VALUE_KEY)) {
+            throw new IllegalArgumentException("Each item must contain 'weight' and 'value' keys");
+        }
+        validateValue(item.get(VALUE_KEY));
+        validateWeight(item.get(WEIGHT_KEY));
+    }
+
+    private void validateValue(Object valueObj) {
+        if (valueObj == null) {
+            throw new IllegalArgumentException("Value cannot be null");
+        }
+    }
+
+    private void validateWeight(Object weightObj) {
+        if (!(weightObj instanceof Double)) {
+            throw new IllegalArgumentException("Weight must be a non-null Double");
+        }
+
+        double weight = (Double) weightObj;
+
+        if (weight <= 0 || Double.isNaN(weight) || Double.isInfinite(weight)) {
+            throw new IllegalArgumentException("Weight must be a positive number and cannot be NaN or infinite");
+        }
+    }
+
+    private double calculateTotalWeight(List<Map<String, Object>> items) {
+        return items.stream()
+            .mapToDouble(item -> (Double) item.get(WEIGHT_KEY))
+            .sum();
+    }
```

Comment:
How will it work if 
1. there are N weights each of them is Double.MAX_VALUE - 1 or something like that, are we protected against overflow?
2. what if there are custom fields with same name however with different meaning in same provider and even number type? Will there at least any warning appear  or how should user be warned about this (without hours of debugging)

Reply:
1. I appreciate your insight regarding potential overflow. While weights are typically not expected to be extremely large, I've implemented an iterative approach to ensure that adding each weight is checked against Double.MAX_VALUE. This helps prevent overflow in cases where unusually large weights might occur.
2. I believe that values should be unique when performing weighted selection. Therefore, while weights can be the same, duplicates in the value parameter trigger an exception that informs the user of the specific duplicate value. This approach seems appropriate to ensure the integrity of the selection process.

Reply:
> While weights are typically not expected to be extremely large,

if we are talking about project in open source which is used then any input should be expected if it passes all existing validation

Reply:
Sure, I got your point. I provided the guard for overflow.